src/index.ts

```diff
@@ -248,6 +256,42 @@ export const supportedV2RouteHandler = (req, res, next, server) => {
   return server(req, res, next)
 }
 
+// Middleware that must be mounted on our GraphQL route _before_
+// our regular GraphQL handler. This is so we can register a new `res.end`
+// method that will intercept the response and check if any email provider
+// rate limting occurred.
+//
+// If so, we return a 429 status code and a JSON error message.
+app.use("/v2", function (req, res, next) {
```

Comment:
If we're gonna chatGPT stuff we should at least follow modern JS conventions 😄 we never see these types of anonymous `function` declarations anywhere, not for years. Please use arrow functions